components/weather_data/WeatherStationDetail.tsx

```diff
@@ -269,6 +259,126 @@ export const orderStationVariables = (stationVariables: Variable[], timezone: st
   return out;
 };
 
+const columnPadding = 3;
+const rowPadding = 2;
+
+function NewWeatherDataTable({columns: columnsWithTime, timeseries}: {columns: columnData[]; timeseries: WeatherStationTimeseries}) {
+  // DataGrid expects data in row-major order, so we need to transpose our data
+  const [times, ...columns] = columnsWithTime;
+  const data: string[][] = useMemo(() => {
+    const columnMajorArray = [...columns.map(column => formatData(column.variable, column.data))];
+    return columnMajorArray[0].map((_, rowIndex) => columnMajorArray.map(column => String(column[rowIndex])));
+  }, [columns]);
+
+  // Passed to DataGrid to render an individual cell
+  const renderCell = useCallback(
+    ({rowIndex, item}: {rowIndex: number; item: string}) => (
+      <Center flex={1} backgroundColor={colorLookup(rowIndex % 2 ? 'light.100' : 'light.300')}>
+        <BodyXSm>{item}</BodyXSm>
+      </Center>
+    ),
+    [],
+  );
+
+  // Passed to DataGrid to render an individual row header (in this case, the time)
+  const renderRowHeader = useCallback(
+    ({item, rowIndex}: {rowIndex: number; item: string}) => (
+      <Center flex={1} backgroundColor={colorLookup(rowIndex % 2 ? 'light.100' : 'light.300')} borderRightWidth={1} borderColor={colorLookup('text.tertiary')}>
+        <BodyXSm>{item}</BodyXSm>
+      </Center>
+    ),
+    [],
+  );
+
+  // Passed to DataGrid to render an individual column header (in this case, the label, units and elevation)
+  const renderColumnHeader = useCallback(
+    ({
+      item: {name, units},
+    }: {
+      item: {
+        name: string;
+        units: string;
+      };
+    }) => (
+      <VStack
+        flex={1}
+        alignItems="center"
+        justifyContent="flex-start"
+        py={rowPadding}
+        px={columnPadding}
+        bg="blue2"
+        borderBottomWidth={1}
+        borderColor={colorLookup('text.tertiary')}>
+        <BodyXSmBlack color="white">{name}</BodyXSmBlack>
+        <BodyXSmBlack color="white">{units}</BodyXSmBlack>
+      </VStack>
+    ),
+    [],
+  );
+
+  // Passed to DataGrid to render the top-left corner cell. Shows the "Time" label and "PST" units
+  const renderCornerHeader = useCallback(
+    () => (
+      <VStack
+        flex={1}
+        alignItems="center"
+        justifyContent="flex-start"
+        py={rowPadding}
+        px={columnPadding}
+        bg="blue2"
+        borderBottomWidth={1}
+        borderColor={colorLookup('text.tertiary')}>
+        <BodyXSmBlack color="white">Time</BodyXSmBlack>
+        <BodyXSmBlack color="white">PST</BodyXSmBlack>
```

Comment:
The previous behavior here was to show the timezone of the user and to localize times - which is also what the website does outside of the old-school NWAC "NOW" views, and what the other centers get on their weather station maps - can we retain that behavior?

Reply:
good callout! see 5801a30